### PR TITLE
docs: add notes about gc spec compliance

### PR DESCRIPTION
**Pull request by ronag**
Refs: https://github.com/nodejs/undici/issues/998

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +166,40 @@ Only supported on Node 16+.
...
+      // force consumption of body
+    }
+    return res.headers
+  })
+
+// Do not
+const headers = await fetch(url)
+  .then(res => res.headers)
+```
+
```
Comment (ronag) [inline README.md:202]: Should we add an alternative solution through `--expose-gc`? i.e. if `global.gc` is available use some internal undici heuristics to manually force garbage collection to release connections?

Comment (ronag) [reply README.md:202]: @addaleax: Thoughts?

Comment (ronag) [reply README.md:202]: e.g. if all `Pool` connections are allocated (i.e. we stall or deadlock) then force a gc.

Comment (ronag) [reply README.md:202]: When moving into core we could achieve this without the runtime flag.

Comment (ronag) [reply README.md:201]: @mcollina

File: README.md [hunk 2]
```diff
@@ -166,6 +166,39 @@ Only supported on Node 16+.
 
 This is [experimental](https://nodejs.org/api/documentation.html#documentation_stability_index) and is not yet fully compliant with the Fetch Standard. We plan to ship breaking changes to this feature until it is out of experimental.
 
+#### Specification Compliance
+
+This section documents parts of the [fetch specification](https://fetch.spec.whatwg.org) which Undici does
+not support or does not fully implement.
+
+##### Garbage collection
```
Comment (ronag) [inline README.md:174]:
```suggestion
##### Garbage Collection
```

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (mcollina) [reply README.md:201]: I don't think adding a `--expose-gc` solution is needed.
The people that would need the automatic cleaning up of resources are the ones that will never pass `--expose-gc`.

I think we should document 1) what error they will be getting and 2) what to do to mitigate that error, e.g. consume the body.

Comment (szmarczak) [reply README.md:201]: > The people that would need the automatic cleaning up of resources are the ones that will never pass --expose-gc.

I second this.

**Pull request merged**